### PR TITLE
primeorder: use `elliptic_curve::point::LookupTable`

### PR DESCRIPTION
**Pull request by Pranavjeet-Naidu**
Fixes #1657
- Add `Radix16Decomposition` in primeorder: generic signed radix-16 digit
  decomposition before the main multiplication loop, matching k256's approach.
- Refactor `ProjectivePoint::mul` and `lincomb` to use
  `elliptic_curve::point::LookupTable` (8-entry signed) instead of the private
  16-entry unsigned table.
- Decode scalars via `Into::<Uint>::into(scalar).to_be_byte_array()` so
  decomposition is correct for LE `FieldBytes` curves (bignp256) and curves
  whose `Uint` is wider than the scalar field (p224, p521).

No new dependencies; radix16 edge cases are covered by unit tests in primeorder
and by existing lincomb proptests in the curve crates.

### CODE REVIEW

File: primeorder/src/radix16.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
+//! Radix-16 signed-digit decomposition for constant-time scalar multiplication.
+
+use elliptic_curve::{CurveArithmetic, PrimeCurve, PrimeField, Scalar, bigint::ArrayEncoding};
+
+/// Worst case: p521 (66-byte scalars) → 132 nibbles + 1 carry digit.
+const MAX_DIGITS: usize = 133;
```
Comment (tarcieri) [inline primeorder/src/radix16.rs]: This size should probably be computed from e.g. `FieldBytesSize`, rather than hardcoded

Comment (tarcieri) [reply primeorder/src/radix16.rs]: FWIW, it's the same size as an uncompressed SEC1 point, though I would probably prefer the `typenum` arithmetic to calculate it be duplicated as opposed to relying on that

Comment (Pranavjeet-Naidu) [reply primeorder/src/radix16.rs]: That does make sense yes, thank you
It will be derived from: 
```rust
<Prod<MaxScalarFieldBytes, U2> as Add<U1>>::Output;
```

File: primeorder/src/radix16.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    array::typenum::{Prod, U1, U2, Unsigned},
+    bigint::ArrayEncoding,
+    consts::U66,
+};
+
+/// Largest scalar `FieldBytes` size among primeorder curves (P-521).
+type MaxScalarFieldBytes = U66;
+
+/// Two nibbles per scalar byte, plus one carry digit for signed recentering.
+type MaxDigits = <Prod<MaxScalarFieldBytes, U2> as Add<U1>>::Output;
```
Comment (tarcieri) [inline primeorder/src/radix16.rs:16]: I meant using `C::FieldBytesSize` to size `digits` in `Radix16Decomposition` on a curve-by-curve basis as opposed to hardcoding a max, but it's okay, I can fix it in a followup.

**Review by tarcieri — APPROVED** [review]
Will do some followups on this, but it gets the basic implementation in place
